stdlib/public/SDK/Foundation/JSONEncoder.swift

```diff
@@ -2005,7 +2005,9 @@ extension _JSONDecoder : SingleValueDecodingContainer {
     }
 
     public func decode<T : Decodable>(_ type: T.Type) throws -> T {
-        try expectNonNull(type)
+        if !_isOptional(type) {
```

Comment:
This is for std lib internal use only. It will probably be made private in Swift 5 (now that we have `@usableFromInline`)

Reply:
@millenomi Suggested taking a look at `_Unwrappable`, which does something similar and is used in bridging. Alternatively, we could introduce the same type of protocol locally here and conform similarly, but we've run into issues with doing something like this in #16238; thoughts?

We can't take the overloading approach because that candidate wouldn't get called — the interface here is fully generic (the exposed interface to calling these methods is the existential `SingleValueDecodingContainer`, not the concrete type).

Reply:
Looks like `_Unwrappable` isn't public, though, so I guess we don't have much of a choice here.